pkg/api/controller.go

```diff
@@ -3006,6 +3012,58 @@ func (c *Controller) GetCurrentUser(w http.ResponseWriter, r *http.Request) {
 	writeResponse(w, http.StatusOK, response)
 }
 
+func (c *Controller) sendResetPasswordEmail(email []string, token string) error {
+	return c.Emailer.SendEmail(email, token, token, []string{})
+}
+
+func (c *Controller) PasswordForgot(w http.ResponseWriter, r *http.Request, body PasswordForgotJSONRequestBody) {
+	user, err := c.Auth.GetUserByEmail(r.Context(), body.Email)
+	if err != nil {
+		writeError(w, http.StatusInternalServerError, err)
+		return
+	}
+	if strings.Compare(*user.Email, body.Email) != 0 {
+		writeError(w, http.StatusNotFound, "No such email")
+		return
+	}
+	secret := c.Auth.SecretStore().SharedSecret()
+	currentTime := time.Now()
+	token, err := GenerateJWT(secret, ResetPasswordAudience, auth.InvalidUserID, body.Email, currentTime, currentTime.Add(tokenExpiryDuration))
+	if err != nil {
+		writeError(w, http.StatusInternalServerError, err)
+		return
+	}
+	// TODO (@shimi9276) create template for sending the email with link for reset
+	err = c.sendResetPasswordEmail([]string{body.Email}, token)
+	if err != nil {
+		writeError(w, http.StatusBadRequest, err)
+		return
+	}
+	writeResponse(w, http.StatusOK, err)
+}
+
+func (c *Controller) SetPassword(w http.ResponseWriter, r *http.Request, body SetPasswordJSONRequestBody) {
+	claims, err := VerifyResetPasswordToken(c.Auth, body.Token)
+	if err != nil {
+		writeError(w, http.StatusForbidden, err)
+		return
+	}
+	if strings.Compare(claims.Id, body.Email) != 0 {
```

Comment:
Can we use == here to compare strings?

Reply:
I don't see the new code, did you forget to push?

Reply:
Sorry, Used it in many places in the code and missed fixing this one.